### ISSUE TITLE
How to calculate custom model metrices (recall and precision)

### ISSUE THREAD

**Issue by kaede-kmr** [kahst/BirdNET-Analyzer#295]
Hello,

I'm interested in how to calculate recall and precision values for custom training models. I'm currently training a custom classifier to identify eight frog species based on the pre-trained BirdNET model. It would be helpful to know the performance metrics (precision and recall) for each of the eight frog species.

What I intended is something like [sklearn.metrics.classification_report](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html), which displays precision and recall for all target classes.

**Example**
```
                precision    recall  f1-score   support

     class 0       0.50      1.00      0.67         1
     class 1       0.00      0.00      0.00         1
     class 2       1.00      0.67      0.80         3

    accuracy                           0.60         5
    macro avg       0.50      0.56      0.49         5
    weighted avg       0.70      0.60      0.61         5
```

While BirdNET provides segments.py for output verification, it doesn't seem to offer information about recall.
Although it might be straitforward to modify the existing train.py script to save the classification report, it was difficult for me to figure out how validation labels and predictions were stored during the training process.

Thank you!

**Comment by Josef-Haupt** [comment]
Generally speaking, you could add your own metrics in [`model.py`](https://github.com/kahst/BirdNET-Analyzer/blob/750c380db094da8acd4954dfd2335b18afe15831/model.py#L258). But we might also add a table after training is complete, with some of the most common metrics. @kahst what do you think?

**Comment by max-mauermann** [comment]
@kahst ist working on an eval-script on this branch: https://github.com/kahst/BirdNET-Analyzer/tree/eval-script to calculate metrics based on a separate test set, which might be what you are looking for.

I am also looking into incorporating this into the training script as well at this branch: https://github.com/kahst/BirdNET-Analyzer/tree/training-with-test-data so that evaluation is done right after the training.

Keep in mind, that performance evaluation should be done on a separate dataset, not the validation split from the training data.

**Comment by Josef-Haupt** [comment]
Closing this issue. We are working on tools to evaluate a custom classifier.
